src/midi-player.ts

```diff
@@ -16,31 +17,109 @@ export class MidiPlayer implements IMidiPlayer {
 
     private _offset: null | number;
 
+    private _latest: null | number;
+
     private _resolve: null | (() => void);
 
     private _scheduler: Scheduler;
 
     private _schedulerSubscription: null | { unsubscribe(): void };
 
+    private _channels: Set<number>;
+
     constructor({ encodeMidiMessage, json, midiFileSlicer, midiOutput, scheduler }: IMidiPlayerOptions) {
         this._encodeMidiMessage = encodeMidiMessage;
         this._endedTracks = null;
         this._json = json;
         this._midiFileSlicer = midiFileSlicer;
         this._midiOutput = midiOutput;
         this._offset = null;
+        this._latest = null;
         this._resolve = null;
         this._scheduler = scheduler;
         this._schedulerSubscription = null;
+
+        // List all channels that are contained in the MIDI file.
+        // We will use it to send All Sound Off messages on pause / stop.
+        this._channels = this._json.tracks.reduce((channels, track) => {
```

Comment:
@chrisguttandin I've seen other players send to all channels 1-16 instead of accumulating channels... What do you think?